oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentPropertyState.java

```diff
@@ -116,7 +163,20 @@ public int count() {
      */
     @NotNull
     String getValue() {
-        return value;
+        return value != null ? value : decompress(this.compressedValue);
+    }
+
+    private String decompress(byte[] value) {
+        try {
+            return new String(compression.getInputStream(new ByteArrayInputStream(value)).readAllBytes(), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            LOG.error("Failed to decompress property {} value: ", getName(), e);
+            return "\"{}\"";
+        }
+    }
+
+    private byte[] getCompressedValue() {
```

Comment:
this method is currently unused but instead the junit class does reflection to access `compressedValue`. If this method was made less protected it could actually replace that reflection use.

Reply:
It actually was my proposal not to expose it. Maybe we can just *copy* or *move* it to the test class?

Reply:
ah, now I remember. Your comment wasn't visible in this particular review anymore, so I didn't connect the dots.

I'm actually a fan of facilitating testing, and that can mean to introduce methods in productive classes that are there solely for testing purpose. That method can still be non-public but package-protected for example. That in my view makes things more explicit than using reflection in tests which is brittle. Moving this method to the test class I assume implies opening up the actual variable, which seems less favourable.

Reply:
Done